google/cloud/security/common/gcp_type/firewall_rule.py

```diff
@@ -195,40 +213,49 @@ def is_equivalent(self, other):
         """
         return (self.direction == other.direction and
                 self.network == other.network and
-                self.source_tags == other.source_tags and
-                self.target_tags == other.target_tags and
+                self._source_tags == other._source_tags and
+                self._target_tags == other._target_tags and
                 self.source_ranges == other.source_ranges and
                 self.destination_ranges == other.destination_ranges and
                 self.firewall_action.is_equivalent(other.firewall_action))
+    # pylint: enable=protected-access
 
 
 class FirewallAction(object):
     """An association of allowed or denied ports and protocols."""
 
-    def __init__(self, firewall_rule_allowed=None, firewall_rule_denied=None):
+    ALLOW = 'allow'
```

Comment:
Since these are only used to verify that firewall_rule_action's value is valid, why not:

VALID_ACTIONS = frozenset(['allow', 'deny'])

Reply:
Done